mlflow/projects/__init__.py

```diff
@@ -724,7 +724,7 @@ def _parse_kubernetes_config(backend_config):
         raise ExecutionException("Could not find 'kube-job-template-path': {}".format(
             kube_job_template))
     if 'kube-context' not in backend_config.keys():
-        raise ExecutionException("Could not find kube-context in backend_config.")
+        _logger.info("Could not find kube-context in backend_config. Using in-cluster config.")
```

Comment:
Nit: Can we make this a warning instead?

Reply:
For the case where this is inside a pod already having this as a warning could be the wrong signal, as I would expect after this change it would be expected behavior.

imho, even outside of kubernetes itself I would expect it to pick the current context when not provided, as someone who switches contexts to switch clusters having to also modify some file is less then ideal

Reply:
@ReDucTor That's a great point. @zanitete is it possible to use the current context if none is specified? Will this also work for in-cluster contexts?

Reply:
The in-cluster configuration does not rely on a config file, so there's no "current context" in that scenario.

We could try first to use the current context (that covers the cases when `KUBECONFIG` is set or `~/.kube/config` exists) and fallback to the in-cluster configuration as last option. Would do you think?

Reply:
@zanitete That sounds great!